electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,401 @@
+        tx.update_signatures(signatures)
+
+class BitBox02_KeyStore(Hardware_KeyStore):
+    hw_type = 'bitbox02'
+    device = 'BitBox02'
+    plugin: 'BitBox02Plugin'
+
+    def __init__(self, d):
+        Hardware_KeyStore.__init__(self, d)
+        # Errors and other user interaction is done through the wallet's
+        # handler.  The handler is per-window and preserved across
+        # device reconnects
+        self.force_watching_only = False
+        self.ux_busy = False
+
+
+    def dump(self):
+        d = Hardware_KeyStore.dump(self)
+        return d
+
+    def get_client(self):
+        # called when user tries to do something like view an address.
+        return self.plugin.get_client(self)
+
+    def give_error(self, message, clear_client=False):
+        self.logger.info(message)
+        if not self.ux_busy:
+            self.handler.show_error(message)
+        else:
+            self.ux_busy = False
+        if clear_client:
+            self.client = None
+        raise UserFacingException(message)
+
+    def sign_transaction(self, tx, password):
+        # Upload PSBT for signing.
+        # - we can also work offline (without paired device present)
+        if tx.is_complete():
+            return
+        client = self.get_client()
+
+        try:
+            try:
+                self.handler.show_message("Authorize Transaction...")
+                client.sign_transaction(self, tx)
+
+            finally:
+                self.handler.finished()
+
+        except BaseException as e:
+            self.logger.exception('')
+            self.give_error(e, True)
+            return
+
+    def show_address(self, sequence, txin_type):
+        client = self.get_client()
+        address_path = self.get_derivation_prefix()[2:] + "/%d/%d"%sequence
+        try:
+            try:
+                self.handler.show_message(_("Showing address ..."))
+                dev_addr = client.show_address(address_path, txin_type)
+                # we could double check address here
+            finally:
+                self.handler.finished()
+        except BaseException as exc:
+            self.logger.exception('')
+            self.handler.show_error(exc)
+
+class BitBox02Plugin(HW_PluginBase):
+    keystore_class = BitBox02_KeyStore
+    #minimum_library = (0, 7, 7)
+
+    DEVICE_IDS = [
+            (0x03eb, 0x2403)
+    ]
+
+    SUPPORTED_XTYPES = ('p2wpkh-p2sh', 'p2wpkh', 'standard')
+
+    def __init__(self, parent, config, name):
+        HW_PluginBase.__init__(self, parent, config, name)
+
+        #self.libraries_available = self.check_libraries_available()
+        # We have t he library in the source tree at the moment
+        self.libraries_available = True
+        #if not self.libraries_available:
+        #    return
+
+        self.device_manager().register_devices(self.DEVICE_IDS)
+
+    def get_library_version(self):
+        return "v4.1.0"
```

Comment:
what's this?

Reply:
It's just a dummy value. I removed it again.